src/store/wallet-connect-v2/wallet-connect-v2.effects.ts

```diff
@@ -0,0 +1,487 @@
+    dispatch(
+      LogActions.info(
+        `[WC-V2/walletConnectV2SubscribeToEvents]: new pending request: ${JSON.stringify(
+          event,
+        )}`,
+      ),
+    );
+    if (
+      Object.keys(WALLET_CONNECT_SUPPORTED_CHAINS).includes(
+        event.params.chainId,
+      )
+    ) {
+      dispatch(
+        WalletConnectV2Actions.sesionRequest({...event, createdOn: Date.now()}),
+      );
+    }
+  });
+  signClient.on('session_delete', data => {
+    dispatch(
+      LogActions.info(
+        `[WC-V2/walletConnectV2SubscribeToEvents]: session disconnected: ${data.topic}`,
+      ),
+    );
+    dispatch(WalletConnectV2UpdateSessions(data.topic));
+  });
+};
+
+export const walletConnectV2ApproveCallRequest =
+  (request: WCV2RequestType, wallet: Wallet): Effect<Promise<void>> =>
+  dispatch => {
+    return new Promise(async (resolve, reject) => {
+      const {topic, id} = request;
+      try {
+        const response = await dispatch(approveEIP155Request(request, wallet));
+        await signClient.respond({
+          topic,
+          response,
+        });
+        dispatch(WalletConnectV2UpdateRequests(id));
+        dispatch(
+          LogActions.info(
+            '[WC-V2/walletConnectV2ApproveCallRequest]: call request approval',
+          ),
+        );
+        await sleep(500);
+        resolve();
+      } catch (err) {
+        dispatch(WalletConnectV2UpdateRequests(id));
+        dispatch(
+          LogActions.error(
+            '[WC-V2/walletConnectV2ApproveCallRequest]: an error occurred while approving call request',
+          ),
+        );
+        dispatch(LogActions.error(JSON.stringify(err)));
+        reject(err);
+      }
+    });
+  };
+
+export const walletConnectV2RejectCallRequest =
+  (request: WCV2RequestType): Effect<Promise<void>> =>
+  dispatch => {
+    return new Promise(async (resolve, reject) => {
+      const {id, topic} = request;
+      try {
+        const response = formatJsonRpcError(
+          id,
+          getSdkError('USER_REJECTED_METHODS').message,
+        );
+        await signClient.respond({
+          topic,
+          response,
+        });
+        dispatch(WalletConnectV2UpdateRequests(id));
+        dispatch(
+          LogActions.info(
+            '[WC-V2/walletConnectV2RejectCallRequest]: call request rejection',
+          ),
+        );
+        resolve();
+      } catch (err) {
+        dispatch(WalletConnectV2UpdateRequests(id));
+        dispatch(
+          LogActions.error(
+            '[WC-V2/walletConnectV2RejectCallRequest]: an error occurred while rejecting call request',
+          ),
+        );
+        dispatch(LogActions.error(JSON.stringify(err)));
+        reject(err);
+      }
+    });
+  };
+
+export const walletConnectV2OnDeleteSession =
+  (topic: string): Effect<Promise<void>> =>
+  dispatch => {
+    return new Promise(async (resolve, reject) => {
+      try {
+        await signClient.disconnect({
```

Comment:
double check but i think this needs to be `await signClient.session.delete(topic, getSdkError('USER_DISCONNECTED'));`